pkg/readiness/ready_tracker_test.go

```diff
@@ -541,41 +550,63 @@ func Test_Tracker(t *testing.T) {
 	}
 }
 
-// Verifies that a Config resource referencing bogus (unregistered) GVKs will not halt readiness.
-func Test_Tracker_UnregisteredCachedData(t *testing.T) {
-	g := gomega.NewWithT(t)
-
-	testutils.Setenv(t, "POD_NAME", "no-pod")
-
-	// Apply fixtures *before* the controllers are setup.
-	err := applyFixtures("testdata")
-	if err != nil {
-		t.Fatalf("applying fixtures: %v", err)
-	}
-
-	// Apply config resource with bogus GVK reference
-	err = applyFixtures("testdata/bogus-config")
-	if err != nil {
-		t.Fatalf("applying config: %v", err)
-	}
-
-	// Wire up the rest.
-	mgr, wm := setupManager(t)
-	cfClient := setupDataClient(t)
-	providerCache := frameworksexternaldata.NewCache()
-
-	if err := setupController(mgr, wm, cfClient, mutation.NewSystem(mutation.SystemOpts{}), nil, providerCache); err != nil {
-		t.Fatalf("setupControllers: %v", err)
+// Verifies additional scenarios to the base "testdata/" fixtures, such as
+// invalid Config resources or overlapping SyncSets, which we want to
+// make sure the ReadyTracker can handle gracefully.
+func Test_Tracker_EdgeCases(t *testing.T) {
+	tts := []struct {
+		name         string
+		fixturesPath string
+	}{
+		{
+			name:         "overlapping syncsets",
+			fixturesPath: "testdata/syncset",
+		},
+		{
+			// bad gvk in Config doesn't halt readiness
+			name:         "bad gvk",
+			fixturesPath: "testdata/config/bad-gvk",
+		},
+		{
+			// repeating gvk in Config doesn't halt readiness
+			name:         "repeating gvk",
+			fixturesPath: "testdata/config/repeating-gvk",
+		},
+		{
+			// empty syncOnly still needs reconciling for a Config resources
```

Comment:
I don't see how this is tested?

If we altered the code such that configs with an empty syncOnly field didn't need reconciling, wouldn't this test still pass?

I'd delete the test, it's not a super interesting edge case given the structure of the code.